styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -648,6 +649,8 @@ private void emitPodEvents(Pod pod, RunState runState) {
       } catch (StateTransitionConflictException e) {
         LOG.debug("State transition conflict on kubernetes pod event: {}", event, e);
         return;
+      } catch (CounterCapacityException e) {
+        LOG.debug("Counter capacity exhausted when processing pod event: {}", event, e);
```

Comment:
Don't we need an early return here as well?

Reply:
NVM after checking the complete code.

Reply:
NVM after checking the complete code.

Reply:
Probably should. Overall this event generation code smells.